<launch>
	<!-- global rosparam -->
	<!-- <param name="use_sim_time" value="true"/> -->
	
  <!-- main -->
	<node pkg="sq_scan_matching" name="matching_ekf" type="matching_ekf">
	<!-- <node pkg="sq_scan_matching" name="ekf" type="ekf" output="screen"> -->
		<remap from="/odom" to="/sq_lidar/odom"/>
		<param name="init_x" type="double" value="0.0"/>
		<param name="init_y" type="double" value="0.0"/>
		<!-- <param name="init_x" type="double" value="-33.083"/>
		<param name="init_y" type="double" value="29.004"/> -->
		<param name="init_z" type="double" value="0.0"/>
		<param name="init_roll" type="double" value="0.0"/>
		<param name="init_pitch" type="double" value="0.0"/>
		<param name="init_yaw" type="double" value="-1.5"/>
		<param name="sigma_imu" type="double" value="1e-5"/>
		<param name="sigma_odom" type="double" value="1e-5"/>
		<param name="sigma_ndt" type="double" value="1e+1"/>
	</node>

	<node pkg="sq_scan_matching" name="icp_sq_scan_matcher" type="icp_sq_scan_matcher">
		<!-- <remap from="/velodyne_points" to="/scanline_life_evaluator/saved_cloud"/> -->
		<remap from="/velodyne_points" to="/cloud/static"/>
		<param name="pc_range" type="double" value="40.0"/>
		<param name="leafsize_source" type="double" value="0.15"/>
		<param name="leafsize_target" type="double" value="0.2"/>
		<param name="trans_epsilon" type="double" value="1e-3"/>
		<param name="stepsize" type="double" value="0.1"/>
		<param name="resolution" type="double" value="0.5"/>
		<param name="max_iterations" type="int" value="30"/>
		<param name="correspondence_distance" type="double" value="1.0"/>
		<param name="matching_score_th" type="double" value="180"/>
	</node>
	
	<!-- <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" output="screen">
			<param name="file_name" value="/home/amsl/pcd/pcd_data/ikuta/ikuta_outdoor.pcd" />
			<param name="frame_id" value="map" />
			<param name="interval" value="3" />
	</node> -->
  <!-- <node pkg="rviz" type="rviz" name="rviz21" args="-d $(find sq_scan_matching)/config/icp.rviz" /> -->
</launch>